staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -419,6 +431,59 @@ type ValidatingAdmissionPolicyBindingSpec struct {
 	// Required.
 	// +listType=set
 	ValidationActions []ValidationAction `json:"validationActions,omitempty" protobuf:"bytes,4,rep,name=validationActions"`
+
+	// NamespaceParamRef specifies the parameter resource or resources used to
+	// configure the admission control policy. It should point to a resources of
+	// the type specified in ParamKind of the bound ValidatingAdmissionPolicy.
+	//
+	// If the policy specifies a ParamKind and the resource referred to by
+	// ParamRef does not exist, this binding is considered mis-configured and
+	// the FailurePolicy of the ValidatingAdmissionPolicy is applied.
+	//
+	// if the paramKind of the policy referred to by `policyName` is cluster scoped,
+	// and namespaceParamRef set, the binding is considered mis-configured, and
+	// the failureMode applies.
+	//
+	// `namespaceParamRef` and `paramRef` are members of a union; if one of the
+	// fields is set, the other must be unset.
+	// +optional
+	NamespaceParamRef *NamespaceParamRef `json:"namespaceParamRef,omitempty" protobuf:"bytes,5,opt,name=namespaceParamRef"`
+}
+
+// NamespaceParamRef references a parameter resource scoped to the same
+// namespace as the resource being evaluated by a policy.
+type NamespaceParamRef struct {
+	// Name is the name of the namespace-scoped param to evaluate against.
+	// When evaluating the admission of a resource request, a param with the
+	// given name will be found in the same namespace as the resource and used
+	// as the `param` input to the policy's rules.
+	//
+	// Exactly one of `name` or `selector` must be specified, but not both.
+	//
+	// +optional
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// Selector can be used to match param resources based on its metadata.
+	// If multiple objects are found to match, the policy will be evaluated
+	// against ALL of them and the result will be ANDed together.
+	// If no objects match the selector, then the `failAction` applies.
```

Comment:
do we have a good sense for the number of evaluations we could be adding for a single request with multiple levels of multiplication (e.g. `x` policies * `y` bindings * `z` params)? how do execution budgets work across all of those evaluations?

Reply:
Good question.  I don't think we do, but it's going to be easier given that param objects are going to be easy to add, so there is clearly some risk here.

I don't know if runtime cost really helps. If we did accumulate cost for a policy*binding then anyone that writes too many params could just create a new binding to bypass the limit.

Is having a better understanding of this be  something we should add to our GA graduation criteria?

Reply:
added some new docs to `ValidatingAdmissionPolicyBinding` about this topic, PTAL.

Reply:
That's a good point! We currently have budget mainly against bindings and I can see issues of having too many validations on a single request if we don't limit it. I am thinking if it would address your concerns by adding limits on the number of bindings based on request matched?

Reply:
EDIT: It should be two different limits on # of policy evals and # of evals per request. (by # of evals per request I mean the limit on creating deployments will not affect the limit on creating ns, etc). And we could have a higher limit set on how many policy evals allowed in total. Not sure if we need both for beta or one of them is good enough to go to beta and leave the other as GA graduation criteria. Thank you for raising it!

Reply:
>  I am thinking if it would address your concerns by adding limits on the number of bindings based on request matched?

I'm trying to think of the behavior if the limits are exceeded... it's pretty weird for creating 1-n params instances to work, and then creating n+1 params instances to either be ignored or make things start failing... :-/

Reply:
Is this OK to address after beta?

Reply:
I think we need agreement on a plan for the behavior at the limits *before* beta, to make sure we don't commit ourselves to something that will ultimately be unsupportable or unbounded

Reply:
What do you think about adding a new field to `ValidatingAdmissionPolicy`: `EvalLimitExceededAction` which chooses the action to take when a per-request CEL cost budget for evaluations is exceeded. User may select `Allow` or `Deny`.

Reply:
I understand simply adding runtime cost limits will not address the concern of what behavior is expected when passing runtime limit. The options for adding runtime cost might be:
1. Fail the request which might cause concerns on control plane availability. And user has to either switch validationAction/failurePolicy to bypass it or refactor the validation(either reduce the size or write more efficient rules).
2. Only return warnings when certain limit exceed.
3. Let user specify the action when certain limit exceed.

I am thinking we could also add limit on how many policy/binding would be allowed to create based on scalability evaluation before GA which will prevent user from easily exceeding the runtime cost limit. Curious on what others think. @jpbetz @liggitt